AMP-61625: Marketing Analytics Browser (#323)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/sdks/typescript-browser/index.md

!!!info "Browser SDK Resources"
    [:material-github: Github](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser) · [:material-code-tags-check: Releases](https://github.com/amplitude/Amplitude-TypeScript/releases) · [:material-book: API Reference](https://amplitude.github.io/Amplitude-TypeScript/)

!!!note "Typescript Broswer SDK versus the Marketing Analytics Browser"

>     The TypeScript Browser SDK is our recommended SDK for most users. It's functionality can easy be extended through the use of plugins.

Reviewer: ```suggestion
    Amplitude recommends the TypeScript Browser SDK for most users. You can extend its functionality using plugins.
```

>     However, if you want web attribution and page view tracking without any additional setup and please use Marketing Analytics Browser instead. It extends the Typescript Browser SDK with build in web attribution and page view tracking. Learn more about [Marketing Analytics Browser SDK](../marketing-analytics-browser/).

Reviewer: ```suggestion
    However, if you want web attribution and page view tracking without extra setup, use the Marketing Analytics Browser instead. It extends the Typescript Browser SDK with built-in web attribution and page view tracking. Learn more about [Marketing Analytics Browser SDK](../marketing-analytics-browser/).
```

!!!info "Browser SDK Resources"
    [:material-github: Github](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser) · [:material-code-tags-check: Releases](https://github.com/amplitude/Amplitude-TypeScript/releases) · [:material-book: API Reference](https://amplitude.github.io/Amplitude-TypeScript/)

!!!note "Typescript Broswer SDK versus the Marketing Analytics Browser"

>     If you need both web attribution and page view tracking. Please use Marketing Analytics Browser instead. It extends the Typescript Browser SDK with build in web attribution and page view tracking. Learn more about [Marketing Analytics Browser SDK](../marketing-analytics-browser/).

Reviewer 2: I think we should call out the the TS SDK is prefered for technical users. Here's an attempt at copy. cc @caseyamp 
```suggestion
    The TypeScript Browser SDK is our recommended SDK for most users. It's functionality can easy be extended through the use of plugins.
    However, if you want web attribution and page view tracking without any additional setup and please use Marketing Analytics Browser instead. It extends the Typescript Browser SDK with build in web attribution and page view tracking. Learn more about [Marketing Analytics Browser SDK](../marketing-analytics-browser/).
```

Author: Thanks for helping me refine the words.

  currentId = 100;
  config?: BrowserConfig;

  /**
   * setup() is called on plugin installation
   * example: client.add(new AddEventIdPlugin());
   */
  async setup(config: BrowserConfig): Promise<undefined> {
     this.config = config;
     return;
  }

  /**
   * execute() is called on each event instrumented
   * example: client.track('New Event');
   */
  async execute(event: Event): Promise<Event> {
    event.event_id = this.currentId++;
    return event;
  }
}

init('API_KEY');
add(new AddEventIdPlugin());
```

##### Web Attribution Enrichment Plugin

You need to download `plugin-web-attribution-browser` package and add the `webAttributionPlugin` before call init method. 

=== "npm"

    ```bash
    npm install @amplitude/plugin-web-attribution-browser
    ```

=== "yarn"

    ```bash
    yarn add @amplitude/plugin-web-attribution-browser
    ```

```ts
add(webAttributionPlugin(client, attribution));

init('API_KEY', configuration);
```

Please check the avaiable configuration [here](../marketing-analytics-browser/#configuration).
Learn more about what [Web Attribution Plugin](../marketing-analytics-browser/#web-attribution) support.

##### Page View Enrichment Plugin

> You need to download `plugin-page-view-tracking-browser` and add the `pageViewTrackingPlugin` before call init method.

Reviewer: ```suggestion
You need to download `plugin-page-view-tracking-browser` and add the `pageViewTrackingPlugin` before calling the init method.
```

=== "npm"

    ```bash
    npm install @amplitude/plugin-page-view-tracking-browser
    ```

=== "yarn"

    ```bash
    yarn add @amplitude/plugin-page-view-tracking-browser
    ```

```ts
add(pageViewTrackingPlugin(client, pageViewTrackingOptions));

init('API_KEY', configuration);
```

> Please check the avaiable configuration [here](../marketing-analytics-browser/#configuration).

Reviewer: ```suggestion
 See the [configuration options](../marketing-analytics-browser/#configuration).
```

> Learn more about what [Page View Plugin](../marketing-analytics-browser/#page-view) support.

Reviewer: ```suggestion
Learn more about what the [Page View Plugin](../marketing-analytics-browser/#page-view) supports .
```
